pelican-gfm/gfmSetup.py

```diff
@@ -63,82 +62,77 @@ def dpkg_packages_installed():
-                     WORKSPACE
-                     ])
-    subprocess.call([
-                     'tar',
-                     'zxf',
-                     WORKSPACE + "/" + gfmVars.VERSION + ".tar.gz",
-                     "-C",
-                     WORKSPACE
-                     ]
-                    )
-
-    BUILDSPACE = gfmVars.WORKSPACE + "/" + "cmark-gfm-" + gfmVars.VERSION + "/build"
-
-    if not os.path.isdir(BUILDSPACE):
-        os.mkdir(BUILDSPACE)
-
-    thing1 = subprocess.Popen([
-                                "cmake",
-                                "-DCMARK_TESTS=OFF",
-                                "-DCMARK_STATIC=OFF",
-                                ".."
-                              ], cwd=BUILDSPACE)
-    thing1.wait()
-
-    thing2 = subprocess.Popen(["make"], cwd=BUILDSPACE)
-    thing2.wait()
-
-    # Move the libcmark.so artifacts in place
-    print "Moving files"
-    subprocess.call([
-                     "mv",
-                     BUILDSPACE + "/src/libcmark-gfm.so." + gfmVars.VERSION,
-                     gfmVars.LIBCMARKLOCATION + "libcmark-gfm.so"
-                    ]
-                    )
-    subprocess.call([
-                     "mv",
-                     BUILDSPACE + "/extensions/libcmark-gfmextensions.so." + gfmVars.VERSION,
-                     gfmVars.LIBCMARKLOCATION + "libcmark-gfmextensions.so"
-                    ]
-                    )
+    else:
+        # Configure the environment if it's not already configured
+
+        with tempfile.TemporaryDirectory() as WORKSPACE:
+            subprocess.call([
+                             "wget",
+                             "--quiet",
+                             gfmVars.ARCHIVES + "/" + gfmVars.VERSION + ".tar.gz",
+                             WORKSPACE,
+                             "-P",
+                             WORKSPACE
+                             ])
+            subprocess.call([
+                             'tar',
+                             'zxf',
+                             WORKSPACE + "/" + gfmVars.VERSION + ".tar.gz",
+                             "-C",
+                             WORKSPACE
+                             ]
+                            )
+
+            BUILDSPACE = WORKSPACE + "/" + "cmark-gfm-" + gfmVars.VERSION + "/build"
+
+            if not os.path.isdir(BUILDSPACE):
+                os.mkdir(BUILDSPACE)
+
+            thing1 = subprocess.Popen([
+                                        "cmake",
+                                        "-DCMARK_TESTS=OFF",
+                                        "-DCMARK_STATIC=OFF",
+                                        ".."
+                                      ], cwd=BUILDSPACE)
+            thing1.wait()
+
+            thing2 = subprocess.Popen(["make"], cwd=BUILDSPACE)
+            thing2.wait()
+
+            # Move the libcmark.so artifacts in place
+            print("Moving files")
+            subprocess.call([
```

Comment:
Is there some reason you can't use `shutil.move` to move the files here?

Reply:
good catch! I ended up going with shutil for these moves.